### PR TITLE
chore: Bump moby version to 3.0.12

### PR DESCRIPTION
**Pull request by cpuguy83**
**Reason for Change**:

New moby patch release

**Issue Fixed**:

N/A

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by mboersma** [comment]
@cpuguy83 we should also bump the `DefaultMobyVersion` const in [`pkg/api/const.go`](https://github.com/Azure/aks-engine/blob/master/pkg/api/const.go#L467).

**Comment by cpuguy83** [comment]
@mboersma Fixed.

File: parts/k8s/kubernetesparams.t [hunk 1]
```diff
@@ -237,6 +237,7 @@
          "3.0.8",
          "3.0.10",
          "3.0.11"
```
Comment (jackfrancis) [inline parts/k8s/kubernetesparams.t:239]: missing comma at the end of "3.0.11"

Comment (cpuguy83) [reply parts/k8s/kubernetesparams.t:239]: :rage1:

Comment (cpuguy83) [reply parts/k8s/kubernetesparams.t]: Fixed

**Pull request merged**